CONTRIBUTING.rst

```diff
@@ -0,0 +1,500 @@
+
+We hightly recommend you run these tests locally to avoid any CI failures.
+Using CCACHE and pre-built host tools is optional, however it speeds up the
+execution time considerably.
+
+
+Coding Style
+************
+
+Use this coding guideline to ensure that your development complies with the
+project's style and naming conventions.
+
+.. _Linux kernel coding style:
+   https://kernel.org/doc/html/latest/process/coding-style.html
+
+
+In general, follow the `Linux kernel coding style`_, with the
+following exceptions:
+
+* Add braces to every ``if`` and ``else`` body, even for single-line code
+  blocks. Use the ``--ignore BRACES`` flag to make *checkpatch* stop
+  complaining.
+* Use hard tab stops. Set the tab width 8 spaces.
+* Break lines at 80 characters.
+* Use spaces instead of tabs to align comments after declarations, as needed.
+* Use C89-style single line comments, ``/*  */``. The C99-style single line
+  comment, ``//``, is not allowed.
+* Use ``/**  */`` for doxygen comments that need to appear in the documentation.
+
+The Linux kernel GPL-licensed tool ``checkpatch`` is used to check coding
+style conformity. Checkpatch is available in the scripts directory. To invoke
+it when committing code, edit your *.git/hooks/pre-commit* file to contain:
+
+.. code-block: bash
+
+    #!/bin/sh
+    set -e exec
+    exec git diff --cached | ${ZEPHYR_BASE}/scripts/checkpatch.pl - || true
+
+Contribution Workflow
+*********************
+
+One general practice we encourage, is to make small,
+controlled changes. This practice simplifies review, makes merging and
+rebasing easier, and keeps the change history clear and clean.
+
+When contributing to the Zephyr Project, it is also important you provide as much
+information as you can about your change, update appropriate documentation,
+and test your changes thoroughly before submitting.
+
+The general GitHub workflow used by Zephyr developers uses a combination of
+command line git commands and browser interaction with GitHub.  As it is with
+git, there are multiple ways of getting a task done.  We'll describe a typical
+workflow here:
+
+.. _Create a Fork of Zephyr:
+   https://github.com/zephyrproject-rtos/zephyr#fork-destination-box
+
+#. `Create a Fork of Zephyr`_
+   to your personal account on GitHub
+
+#. On your development computer, clone the fork you just made::
+
+     $ git clone https://github.com/<your github id>/zephyr
+
+   This would be a good time to let git know about the upstream repo too::
+
+     $ git remote add upstream https://github.com/zephyrproject-rtos/zephyr.git
+
+   and verify the remote repos::
+
+     $ git remote -v
+
+#. Create a topic branch (off of master) for your work (if you're addressing
+   Jira issue, we suggest including the Jira issue number in the branch name)::
+
+     $ git checkout master
+     $ git checkout -b fix_comment_typo
+
+   Some Zephyr subsystems do development work on a separate branch from
```

Comment:
It would be good to have a list of these subsystems, and their policies, definitively documented somewhere. I've observed this can be a stumbling block for beginners.